packages/components/data-table/src/data-table.story.js

```diff
@@ -238,7 +238,7 @@ const ColumnConfigForm = (props) => {
               onChange={formik.handleChange}
               value={formik.values.align}
             >
-              <option value={undefined}>default</option>
```

Comment:
I was dumb and didn't realize that setting `value` to undefined is the same as it not being there, so the passed value would be the string `default`, which isn't a valid value for alignment and wouldn't do anything.